source/slang/slang-check-overload.cpp

```diff
@@ -1432,9 +1449,23 @@ namespace Slang
             auto type = DeclRefType::create(m_astBuilder, genericTypeParamDeclRef);
             AddTypeOverloadCandidates(type, context);
         }
+        else if( auto localDeclRef = item.declRef.as<ParamDecl>() )
+        {
+            // We could probably be broader than just parameters here
+            // eventually.
+            // Limit it for now though to make the specialization easier
+            ensureDecl(localDeclRef, DeclCheckState::CanUseFuncSignature);
```

Comment:
I have a feeling that this might need to be `TypesFullyResolved`.
But that is mostly because our current checking of `ThisType` is broken and can't be fixed without a total revamp of DeclRef.

Reply:
Thank :)

Making this change doesn't break any of the tests, so I'll include it. Do you have a test in mind which would fail without this change?